File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        
        // 1. 문자열 배열 사전순 정렬
        Arrays.sort(strs);
        
        // 2. 가장 첫 번째와 마지막 문자열 가져오기
        String s1 = strs[0];
        String s2 = strs[strs.length - 1];
        
        int idx = 0;
        // 3. 두 문자열의 처음부터 글자 비교
        while (idx < s1.length() && idx < s2.length()) {
            if (s1.charAt(idx) == s2.charAt(idx)) {
                idx++;
            } else {
                break;
            }
        }
        
        // 4. 일치하는 길이만큼 잘라서 반환
        return s1.substring(0, idx);
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
     if (strs === null || strs.length === 0) return "";
    
    // 첫 번째 문자열을 기준(prefix)으로 잡습니다.
    let prefix = strs[0];
    
    for (let i = 1; i < strs.length; i++) {
        // prefix가 현재 문자열(strs[i])의 시작 부분과 일치하지 않는 동안,
        // prefix의 마지막 글자를 하나씩 제거합니다.
        while (strs[i].indexOf(prefix) !== 0) {
            prefix = prefix.substring(0, prefix.length - 1);
            
            // 만약 prefix가 빈 문자열이 되면 공통 접두사가 없는 것입니다.
            if (prefix === "") return "";
        }
    }
    
    return prefix;
};